src/lib.rs

```diff
@@ -2945,27 +2945,37 @@ impl EncodingDetector {
     /// to lower-case it. Full DNS label validation is intentionally not performed
     /// to avoid panics when the reality doesn't match the specs.)
     pub fn guess(&self, tld: Option<&[u8]>, allow_utf8: bool) -> &'static Encoding {
+        self.guess_assess(tld, allow_utf8).0
+    }
+
+
+    /// Same as `guess()`, but also returns a Boolean indicating
+    /// whether the guessed encoding had a higher score than at least
+    /// one other candidate. If this method returns `false`, the
+    /// guessed encoding is likely to be wrong.
+    pub fn guess_assess(&self, tld: Option<&[u8]>, allow_utf8: bool) -> (&'static Encoding, bool) {
         let mut tld_type = tld.map_or(Tld::Generic, |tld| {
             assert!(!contains_upper_case_period_or_non_ascii(tld));
             classify_tld(tld)
         });
 
         if self.non_ascii_seen == 0
             && self.esc_seen
-            && self.candidates[Self::ISO_2022_JP_INDEX].score.is_some()
         {
-            return ISO_2022_JP;
+            if self.candidates[Self::ISO_2022_JP_INDEX].score.is_some() {
```

Comment:
What's the purpose of moving the `self.candidates[Self::ISO_2022_JP_INDEX].score.is_some()` part to a new inner `if`?

Reply:
Thanks for the review! I added this because the test I added below (`test_en_windows1252`) was not getting the correct encoding (I remember it was some Asian encoding, so probably ISO-2022-JP).

Reply:
AFAICT, the introduction of an inner `if` makes no logical difference. Also, stylistically, the `return` statement should have a semicolon. In order to avoid obfuscating what's being changed by this changeset, please change the `if` block to:

```
        if self.non_ascii_seen == 0
            && self.esc_seen
            && self.candidates[Self::ISO_2022_JP_INDEX].score.is_some()
        {
            return (ISO_2022_JP, true);
        }
```

Reply:
Yes, sorry, I misread the diff presented by GitHub.

Reply:
The newly force-pushed revision seems to retain the problem.

Reply:
Indeed, sorry about that.